docs/topics/sealed-classes.md

```diff
@@ -1,99 +1,350 @@
 
 [Learn more about the hierarchical structure of multiplatform projects](multiplatform-hierarchy.md). 
 
-## Sealed classes and when expression
+## Use sealed classes with when expression
 
 The key benefit of using sealed classes comes into play when you use them in a [`when`](control-flow.md#when-expression)
-expression. 
-If it's possible to verify that the statement covers all cases, you don't need to add an `else` clause to the statement:
+expression.
+The `when` expression, used with a sealed class, allows the Kotlin compiler to check exhaustively that all possible cases are covered. 
+In such cases, you don't need to add an `else` clause. Here's an example demonstrating this:
 
 ```kotlin
+// Sealed class and its subclasses
+sealed class Error {
+    class FileReadError(val file: String): Error()
+    class DatabaseError(val source: String): Error()
+    object RuntimeError : Error()
+}
+
+//sampleStart
+// Function to log errors
 fun log(e: Error) = when(e) {
-    is FileReadError -> { println("Error while reading file ${e.file}") }
-    is DatabaseError -> { println("Error while reading from database ${e.source}") }
-    is RuntimeError ->  { println("Runtime error") }
-    // the `else` clause is not required because all the cases are covered
+    is Error.FileReadError -> println("Error while reading file ${e.file}")
+    is Error.DatabaseError -> println("Error while reading from database ${e.source}")
+    Error.RuntimeError -> println("Runtime error")
+    // No `else` clause is required because all the cases are covered
+}
+//sampleEnd
+
+// List all errors
+fun main() {
+    val errors = listOf(
+        Error.FileReadError("example.txt"),
+        Error.DatabaseError("usersDatabase"),
+        Error.RuntimeError
+    )
+
+    errors.forEach { log(it) }
 }
 ```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.5"}
 
-> `when` expressions on [`expect`](multiplatform-expect-actual.md) sealed classes in the common code of multiplatform projects still 
-> require an `else` branch. This happens because subclasses of `actual` platform implementations aren't known in the 
-> common code.
+> In multiplatform projects, if you have a sealed class with a `when` expression as an 
+> [expected declaration](multiplatform-expect-actual.md) in your common code, you still need an `else` branch. 
+> This is because subclasses of `actual` platform implementations may extend sealed classes that 
+> aren't known in the common code.
 >
 {type="note"}
+
+## Use case scenarios
+
+Let's explore some practical scenarios where the application of sealed classes or interfaces can be particularly helpful.
+
+### State management in UI applications
+
+You can use sealed classes to represent different states of a UI in an application. 
+This approach allows for structured and safe handling of UI changes. 
+This example demonstrates how to manage various UI states:
```

Comment:
Do we need this sentence or it could be omited?

Reply:
I don't think it's necessary, somewhat eases the transition from explanation to example, but I don't think it's necessary.

Reply:
Up to you.